contracts/adapters/IPRebalanceExtension.sol

```diff
@@ -0,0 +1,555 @@
+    function transformRemaining(
+        address _transformComponent,
+        bytes memory _transformData
+    )
+        external
+        onlyAllowedCaller(msg.sender)
+    {
+        address underlying = transformComponentInfo[_transformComponent].underlyingComponent;
+        uint256 underlyingUnitsRaw = targetUnitsUnderlying[underlying];
+        require(underlyingUnitsRaw == 0, "raw underlying in target set composition");
+
+        _transform(_transformComponent, _transformData, true);
+    }
+
+    /* ======== Internal Functions ======== */
+
+    /**
+     * Untransforms a component. To prevent losses from untransforms that may incur slippage, a maximum
+     * untransform size and minimum untransform delay are enforced. 
+     * 
+     * To calculate the number of units to untransform use:
+     * max(currentUnits - targetUnitsUnderlying * exchangeRate, 0)
+     *
+     * This calculation ensures that if the target units is greater than the current units then no tokens will
+     * be untransformed. If this calculation yields a resulting amount to untransform that is larger than the
+     * maximum untransform size, the untransform size will be replaced with this max value. The rest of the
+     * transform component can be untransformed in later transactions.
+     */
+    function _untransform(address _transformComponent, bytes memory _untransformData) internal {
+        require(stage == RebalanceStage.UNTRANSFORM, "must be in untransform stage");
+
+        TransformInfo memory transformInfo = transformComponentInfo[_transformComponent];
+
+        require(transformInfo.minTransformDelay + lastTransform[_transformComponent] < block.timestamp, "delay not elapsed");
+        lastTransform[_transformComponent] = block.timestamp;
+
+        require(transformInfo.underlyingComponent != address(0), "nothing to untransform");
+        require(
+            transformInfo.transformHelper.shouldUntransform(transformInfo.underlyingComponent, _transformComponent),
+            "untransform unavailable"
+        );
+
+        uint256 targetUnderlying = targetUnitsUnderlying[_transformComponent];
+        uint256 currentUnits = setToken.getDefaultPositionRealUnit(_transformComponent).toUint256();
+
+        // convert target units from underlying to transformed amounts
+        uint256 exchangeRate = transformInfo.transformHelper.getExchangeRate(transformInfo.underlyingComponent, _transformComponent);
+        uint256 targetUnitsInTransformed = targetUnderlying.preciseMul(exchangeRate);
+        uint256 unitsToUntransform = currentUnits > targetUnitsInTransformed ? currentUnits.sub(targetUnitsInTransformed) : 0;
+        
+        uint256 maxUntransformUnits = transformInfo.maxTransformSize.preciseDiv(setToken.totalSupply()).preciseMul(exchangeRate);
+        if (unitsToUntransform > maxUntransformUnits) {
+            unitsToUntransform = maxUntransformUnits;
+        }
+
+
+        require(unitsToUntransform > 0, "nothing to untransform");
```

Comment:
We should not revert here. We should just return back on line `333` if `currentUnits <  targetUnitsInTransformed`, which is the case when `unitrsToUntransform` would be zero. Also, the reason we should not revert is it might break the logic in `batchUntransform`.